monai/bundle/utils.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import re
+from distutils.util import strtobool
+from typing import Any, Dict, Optional, Sequence, Tuple, Union
+
+from monai.bundle.config_parser import ConfigParser
+from monai.utils import ensure_tuple, optional_import
+
+yaml, _ = optional_import("yaml")
+
+
+def load_config_file(filepath: str, **kwargs):
+    """
+    Load config file with specified file path.
+    Suppprt JSON and YAML formats.
+
+    Args:
+        filepath: path of target file to load, supported postfixes: `.json`, `yml`, `yaml`.
+        kwargs: other arguments for `json.load` or `yaml.load`, depends on file format.
+            for more details, please check:
+            https://docs.python.org/3/library/json.html#json.load.
+            https://pyyaml.org/wiki/PyYAMLDocumentation.
+
+    """
+    with open(filepath) as f:
+        if filepath.lower().endswith(".json"):
+            return json.load(f, **kwargs)
+        if filepath.lower().endswith((".yml", ".yaml")):
+            if "Loader" not in kwargs:
+                kwargs["Loader"] = yaml.FullLoader
+            return yaml.load(f, **kwargs)
+        raise ValueError("only support JSON or YAML config file so far.")
+
+
+def load_config_file_content(path: str, **kwargs):
+    """
+    Load part of the content from a config file with specified `id` in the path.
+    Suppprt JSON and YAML formats file.
+
+    Args:
+        path: path of target file to load, it can only load part of it appending target `id`
+            in the path with "#" mark. for example: `/data/config.json`, `/data/config.json#net#<args>`.
+        kwargs: other arguments for `json.load` or `yaml.load`, depends on file format.
+            for more details, please check:
+            https://docs.python.org/3/library/json.html#json.load.
+            https://pyyaml.org/wiki/PyYAMLDocumentation.
+
+    """
+    pattern = r"(json|yaml|yml)"
+    result = re.findall(pattern, path, re.IGNORECASE)
+    if len(result) != 1:
+        raise ValueError(f"path should only contain 1 file, but got: {path}.")
+
+    # split the path into filepath and target id of the content
+    paths = path.split(result[0])
+    parser = ConfigParser(config=load_config_file(paths[0] + result[0], **kwargs))
+    return parser[paths[1][1:] if paths[1] != "" else ""]
+
+
+def parse_id_value(pair: str) -> Tuple[str, Any]:
```

Comment:
Maybe add another function `parse_id_dict` to use where a dictionary is being created:

```python
def parse_id_dict(pairs: Optional[Sequence[str]]) -> Dict[str, Any]:
    if pairs is None:
        return {}
    return dict(map(parse_id_value, pairs))
```

Reply:
Good point, added in the latest commit.

Thanks.